sympy/ntheory/residue_ntheory.py

```diff
@@ -72,6 +72,8 @@ def _primitive_root_prime_iter(p):
 
     """
     # it is assumed that p is an int
+    if isprime(p) is False:
+        raise NotImplementedError("Not implemented for composite p")
```

Comment:
Primitive roots exist also for some composite moduli (e.g. odd prime powers). The implementation below will take care of those cases as well. So these two lines should be omitted.